### PR TITLE
feat: evaluate 1on1 conversation single user [WPB-5806]

### PR DESCRIPTION
**Pull request by markusfassbender**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5806" title="WPB-5806" target="_blank">Sub-taskWPB-5806</a>  [iOS] Implement the logic to evaluate a 1:1 conversation with a single user
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

- See ticket about the trigger
- Copying messages from the old conversation to the new one, is not handled yet. I will create a new ticket for it.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

#### How to Test

Testing is quite difficult, because this is an edge case that should only work as safety network if all other migrations before have failed.

----

#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]
Looks good to me

File: wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneResolvedState.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+public enum OneOnOneConversationResolvedState {
```
Comment (David-Henner) [inline wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneResolvedState.swift]: naming suggestion
```suggestion
public enum OneOnOneConversationResolution {
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift [hunk 2]
```diff
@@ -364,6 +366,50 @@ final class ConversationViewController: UIViewController {
         updateRightNavigationItemsButtons()
     }
 
+    // MARK: Resolve 1-1 conversations
+
+    private func resolveConversationIfOneOnOne() {
+        guard
+            conversation.messageProtocol != .mls,
+            conversation.conversationType == .oneOnOne,
```
Comment (David-Henner) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift]: If a conversation is a "fake 1:1" (i.e: a group conversation with only two participants), I think it will be filtered out here. Perhaps we should check if the conversation is of type `oneOnOne` **or** if it's a "fake 1:1"

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift [hunk 3]
```diff
@@ -364,6 +366,50 @@ final class ConversationViewController: UIViewController {
...
+
+        Task {
+            do {
+                let resolvedState = try await service.resolveOneOnOneConversation(with: otherUserID, in: context)
+
+                if case .migratedToMLSGroup(let identifier) = resolvedState {
+                    await navigateToNewMLSConversation(mlsGroupIdentifier: identifier, in: context)
+                }
+            } catch {
+                WireLogger.conversation.warn("resolve non MLS 1-1 conversation failed: \(error)")
```
Comment (David-Henner) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift]:
```suggestion
                WireLogger.conversation.warn("resolution of non MLS 1-1 conversation failed: \(error)")
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift (hunk 2 above)
Comment (markusfassbender) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.swift]: Uh so I missed the case of proteus fake 1-1 conversations, that are groups with 2 participants, right? updated the logic, please double check 🙃

**Review by David-Henner — APPROVED** [review]
